src/vs/workbench/parts/debug/browser/linkDetector.ts

```diff
@@ -0,0 +1,107 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import strings = require('vs/base/common/strings');
+import uri from 'vs/base/common/uri';
+import { isMacintosh } from 'vs/base/common/platform';
+import * as errors from 'vs/base/common/errors';
+import { IMouseEvent, StandardMouseEvent } from 'vs/base/browser/mouseEvent';
+import * as nls from 'vs/nls';
+import { IWorkbenchEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { IWorkspaceContextService } from 'vs/platform/workspace/common/workspace';
+
+export class LinkDetector {
+	private static FILE_LOCATION_PATTERNS: RegExp[] = [
+		// group 0: the full thing :)
+		// group 1: absolute path
+		// group 2: drive letter on windows with trailing backslash or leading slash on mac/linux
+		// group 3: line number
+		// group 4: column number
+		// eg: at Context.<anonymous> (c:\Users\someone\Desktop\mocha-runner\test\test.js:26:11)
+		/(?![\(])(?:file:\/\/)?((?:([a-zA-Z]+:)|[^\(\)<>\'\"\[\]:\s]+)(?:[\\/][^\(\)<>\'\"\[\]:]*)?\.[a-zA-Z]+[0-9]*):(\d+)(?::(\d+))?/g
+	];
+
+	constructor(
+		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
+		@IWorkspaceContextService private contextService: IWorkspaceContextService
+	) {
+		// noop
+	}
+
```

Comment:
Since this handleLinks is now a public method we could try to make the API a bit nicer, a personally find it ugly that it returns an HTMLElement or a string. What does that mean?
At least provide some comments to clarify each case, or try to always return an object with the corresponding fields.